app/controllers/application_controller.rb

```diff
@@ -91,6 +91,12 @@ def current_user
       # Ensures no code will use old @current_user info. Treat the user
       # as anonymous (until the login process sets @current_user again):
       @current_user = nil
+
+    end
+
+    cookies.signed["user_token"] = nil
+    if @current_user
+      cookies.signed["user_token"] = @current_user.persistence_token
```

Comment:
Cool, so does this update the cookie on any usage of `current_user`, or when exactly? Thanks!

Reply:
Once this is answered, I think we are good to merge, yes!!!

Reply:
Hi, I am using current_user cookie to let the WebSocket while connecting to know that from which user the WebSocket is connecting so that we can do role-based functionality.

Reply:
Because session is not available in ActionCable channels so while loading page we check for current_user and when we are checking for current_user we set the cookie so that the websocket can let the action cable know which user it is.

Reply:
@jywarren Does this helps?

Reply:
Yep, just want to know, when logging out, will the cookie then be flushed?

Reply:
Can you add any test for this, perhaps a system test? We can do it in follow-up, but I'd like to protect this with some tests.

Reply:
I thought that `current_user` function will be called when we log out and the browser redirect it to any page. But it good to specifically set the cookie while logging out.
Doing it.
Thanks!

Reply:
And regarding System Test, yes we can do this in the follow up pr.

Reply:
I guess it is not needed to set the cookie to nil while logging out because we change the persistence token of the user while logging out.
https://github.com/publiclab/plots2/blob/7b55c883e79d33569482f9a32389914e509705d0/app/controllers/user_sessions_controller.rb#L213
What do you say? I guess it is not needed. Let me know what do you think?